File: backend/transaction-service/transaction-domain/transaction-application-service/src/main/java/com/saver/system/transaction/service/domain/ports/output/repository/CashTransactionRepository.java
package com.saver.system.transaction.service.domain.ports.output.repository;

import com.saver.system.domain.valueobject.UserId;
import com.saver.system.transaction.service.domain.valueobject.TransactionStrategy.CashTransaction;

import java.util.Optional;

public interface CashTransactionRepository {
    CashTransaction save(CashTransaction transaction);

    Optional<CashTransaction> findByUserId(UserId userId);
}

File: old backend - node/services/transactions/src/data-access/repository/cash-transaction-repository.ts
import { injectable, decorate, inject, unmanaged, optional } from "inversify";
import { DataSource, Repository } from "typeorm";
import { TYPES } from "../../container/types/inversify-types";
import { ITransactionRepository } from "../../domain/domain-application-service/ports/output/repository/itransaction-repository";
import { CashTransaction } from "../../domain/domain-core/entity/cash-transaction";
import { CashTransactionEntity } from "../entity/cash-transaction";

decorate(injectable(), Repository)

@injectable()
export class CashTransactionRepository
  extends Repository<CashTransactionEntity>
  implements ITransactionRepository
{
  constructor(@optional() private dataSource: DataSource) {
    super(CashTransactionEntity, dataSource.createEntityManager());
  }

  async createCashTransaction(
    transaction: CashTransaction
  ): Promise<CashTransactionEntity> {
      return await this.save({ ...transaction });
  }
}